src/mon/Monitor.cc

```diff
@@ -4707,6 +4715,72 @@ int Monitor::print_nodes(Formatter *f, ostream& err)
   return 0;
 }
 
+void Monitor::handle_statfs(MonOpRequestRef op)
+{
+  op->mark_pgmon_event(__func__);
+  auto statfs = static_cast<MStatfs*>(op->get_req());
+  auto session = statfs->get_session();
+  if (!session)
+    return;
+  if (!session->is_capable("pg", MON_CAP_R)) {
+    dout(0) << "MStatfs received from entity with insufficient privileges "
+            << session->caps << dendl;
+    return;
+  }
+  if (statfs->fsid != monmap->fsid) {
+    dout(0) << __func__ << " on fsid " << statfs->fsid
+            << " != " << monmap->fsid << dendl;
+    return;
+  }
+  dout(10) << __func__ << " " << *statfs
+           << " from " << statfs->get_orig_source() << dendl;
+  epoch_t ver = 0;
+  if (pgservice == mgrstatmon()->get_pg_stat_service()) {
+    ver = mgrstatmon()->get_last_committed();
+  } else {
+    ver = pgmon()->get_last_committed();
+  }
+  auto reply = new MStatfsReply(monmap->fsid, statfs->get_tid(), ver);
+  reply->h.st = pgservice->get_statfs();
+  send_reply(op, reply);
+}
+
+void Monitor::handle_get_pool_stat(MonOpRequestRef op)
+{
+  op->mark_pgmon_event(__func__);
+  auto m = static_cast<MGetPoolStats*>(op->get_req());
+  auto session = m->get_session();
+  if (!session)
+    return;
+  if (!session->is_capable("pg", MON_CAP_R)) {
+    dout(0) << "MGetPoolStats received from entity with insufficient caps "
+            << session->caps << dendl;
+    return;
+  }
+  if (m->fsid != monmap->fsid) {
+    dout(0) << __func__ << " on fsid "
+	    << m->fsid << " != " << monmap->fsid << dendl;
+    return;
+  }
+  epoch_t ver = 0;
+  if (pgservice == mgrstatmon()->get_pg_stat_service()) {
+    ver = mgrstatmon()->get_last_committed();
+  } else {
+    ver = pgmon()->get_last_committed();
+  }
+  auto reply = new MGetPoolStatsReply(m->fsid, m->get_tid(), ver);
+  for (const auto& pool_name : m->pools) {
+    const auto pool_id = osdmon()->osdmap.lookup_pg_pool_name(pool_name);
+    if (pool_id == -ENOENT)
+      continue;
+    auto pool_stat = pgservice->get_pool_stat(pool_id);
+    if (!pool_stat)
+      continue;
+    reply->pool_stats[pool_name] = *pool_stat;
+  }
+  send_reply(op, reply);
```

Comment:
@tchaikov, we've managed to keep the RADOS cluster stuff out of Monitor.cc so far; do you think we can put these stat handlers in OSDMonitor?